uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-requests-heatmap.component.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-env amd */
+
+define([
+  'common/moment',
+  'leave-absences/shared/modules/components'
+], function (moment, components) {
+  components.component('leaveRequestsHeatmap', {
+    bindings: {
+      leaveRequests: '<'
+    },
+    controller: LeaveRequestsHeatmapController,
+    controllerAs: 'LeaveRequestsHeatmap',
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-requests-heatmap.html';
+    }]
+  });
+
+  function LeaveRequestsHeatmapController () {
+    var vm = this;
+
+    vm.$onChanges = $onChanges;
+
+    /**
+     * Implemenents the $onChanges method for Angular controllers.
+     * When leaveRequests are bound, it maps them to heat map values.
+     *
+     * @param {OnChangesObject} changes - it has the previous and current value
+     * for each bindings change. This value is passed by angular.
+     */
+    function $onChanges (changes) {
+      if (changes.leaveRequests) {
+        mapLeaveRequestsToHeatmapValues();
+      }
+    }
+
+    /**
+     * Stores the total leave balance for each day of the week.
+     */
+    function mapLeaveRequestsToHeatmapValues () {
+      vm.heatmapValues = {};
```

Comment:
I need to add `vm.heatmapValues = {}` to the component definition (before on init). But this line needs to stay because it clears previous heat map values in case the leave request bindings are updated.